clash-lib/src/Clash/Util.hs

```diff
@@ -447,12 +366,6 @@ allM p (x:xs) = do
   else
     return False
 
-traceWith :: (a -> String) -> a -> a
-traceWith f a = trace (f a) a
-
-traceShowWith :: Show b => (a -> b) -> a -> a
-traceShowWith f a = trace (show (f a)) a
-
```

Comment:
These are super useful when debugging!

Reply:
Fair point, what about a debugging module that exports these (e.g. `Clash.Internal.Debug`)?

Reply:
The nice thing about these is that they're usually imported by default as many modules import `Clash.Util` unqualified (eww..).